### ISSUE TITLE
[POTENTIAL BUG][?Or wrong implementation from my side?]Multiple transactions send in a short time frame, only 1 send to wallet

### ISSUE THREAD

**Issue by BlackTurtle123** [jansenmarc/WavesGatewayFramework#12]
I am not sure yet what is the case, but it could be a potential bug. 
When a users spams a few transactions to the gateway, it only picks up one.
Not sure what is the cause/ what part of the gateway is making this happen.
This is the generated addr from the user for reference.
https://explorer.blackturtle.eu/address/3JuYgZNMmbMpuCUXyfxRQiAFL54rdciJEqr

EDIT:
The framework seems to "forget" tx is fhey are fast after each other.
On the following link it get's 6 tx and only send out 3:
Waves:     https://wavesdesk.com/account/3PJ33ZasLyhLaFuK4gHGTFSQbpFBP5Entor
TN:           https://explorer.blackturtle.eu/address/3JqAYiRnuiJxdMVmdTUsxuTV39LXHR5JWXk

**Comment by BlackTurtle123** [comment]
https://gist.github.com/BlackTurtle123/f848d960b876da68c96917eaa04a3f8b

Seems to me, it just doesn't registers all tx?

**Comment by BlackTurtle123** [comment]
This gist contain the logs from what is happening when 6 tx should go to the mainnetwork.
There are only 2 done. 4 aren't noticed.
https://gist.github.com/BlackTurtle123/203293c17cebc0a606e1f599aeaf2bf2

**Comment by henning96** [comment]
The problem is that the attempts for your side (coin) are containing two receiver addresses which is not possible for a Waves transaction (as far as I know...). 
That is because half of your transactions are missing.
In your case, you have to set the property *only_one_transaction_receiver* to *True*.

**Comment by BlackTurtle123** [comment]
And what about not all tx being registered?
If is that the same bug?

**Comment by henning96** [comment]
The problem may be that the Waves block is checked before it is finally completed.
You can check if that is the case if you manually set the waves_block_height in the database to an earlier value than your transactions. If they are registered now, this may be the problem.
The waves_block_height can be found in the key_value collection.

**Comment by jansenmarc** [comment]
The framework should check for new blocks height-1. Does it? Same should apply for the implementation of the fork.

**Comment by BlackTurtle123** [comment]
Even when I set in the config, it needs to wait 1 block before checking it?
           coin_last_block_distance=1,
            waves_last_block_distance=1,

I will now see if something changes when changing block height

**Comment by henning96** [comment]
Yes, it does (it should).

**Comment by henning96** [comment]
@BlackTurtle123 If you set coin_last_block_distance to 1, there should not be any further steps required.

**Comment by BlackTurtle123** [comment]
@henning96 I do think it is sending a lot of tx (and lost fees from the past) atm.

**Comment by henning96** [comment]
Increasing the 1 distance should not be necessary and would not explain the problem. So, please first test if it now works if you reset the height and we will see about further steps.

**Comment by BlackTurtle123** [comment]
So the framework is synced, all tx processed in both directions

**Comment by BlackTurtle123** [comment]
By the spam test this time I see many and many errors pop up like those
`Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 127, in _run
    self._iteration()
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 217, in _iteration
    self._handle_transactions(filtered_transactions)
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 174, in _handle_transactions
    handle_task_group.join(raise_error=True)
  File "/usr/local/lib/python3.5/dist-packages/gevent/pool.py", line 557, in join
    greenlet._raise_exception()
  File "/usr/local/lib/python3.5/dist-packages/gevent/greenlet.py", line 159, in _raise_exception
    reraise(*self.exc_info)
File "/usr/local/lib/python3.5/dist-packages/gevent/_compat.py", line 33, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.5/dist-packages/gevent/greenlet.py", line 536, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 172, in <lambda>
    handle_task_group.spawn(lambda: self._handle_transaction(tr))
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 154, in _handle_transaction
    raise ex
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/transaction_polling_service.py", line 150, in _handle_transaction
    self._consumer.handle_transaction(transaction)
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/coin_transaction_consumer_impl.py", line 202, in handle_transaction
    self._handle_transaction_receiver(transaction.tx, transaction.receivers[i], i, transaction.senders)
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/service/coin_transaction_consumer_impl.py", line 178, in _handle_transaction_receiver
    self._attempt_list_storage.safely_save_attempt_list(attempt_list)
  File "/usr/local/lib/python3.5/dist-packages/waves_gateway/storage/transaction_attempt_list_storage.py", line 72, in safely_save_attempt_list
    raise DuplicateAttemptListTriggerError()
waves_gateway.common.exceptions.DuplicateAttemptListTriggerError`

**Comment by BlackTurtle123** [comment]
For future reference, added second db dump.
15 tx from waves to tn => 11 arrive
7tx from tn to waves => 2 arrive
block distance is at 2 for both coin and waves
https://gist.github.com/BlackTurtle123/203293c17cebc0a606e1f599aeaf2bf2

**Comment by henning96** [comment]
I hope I am not right, but please adjust the _handle_transactions method in the TransactionPollingService class of the Framework and replace it with this:
```
    def _handle_transactions(self, transactions: List[Transaction]) -> None:
        for tr in transactions:
            self._handle_transaction(tr)
```
This is an internal problem and definitely a bug.

**Comment by BlackTurtle123** [comment]
@henning96 first time ever all tx were registered and were send to the owner

**Comment by henning96** [comment]
This should be fixed in Release v1.0.2
